### PR TITLE
Service startup parameters & recover

### PR DESCRIPTION
**Pull request by mhertis**
Startup parameters supported only on Windows...

### CODE REVIEW

**Comment by nehmebilal** [comment]
Can you please provide more context?

File: Samples/Etg.Yams.ARM/Etg.Yams.Host/Program.cs [hunk 1]
```diff
@@ -104,4 +114,50 @@ public static class YamsConfigBuilderExtensions
...
+                {
+
+                    var arguments = Environment.GetCommandLineArgs();
+
+                    string programName = null;
+                    StringBuilder argumentsList = new StringBuilder();
+
+                    for (int i = 0; i < arguments.Length; i++)
+                    {
+                        if (i == 0)
```
Comment (nehmebilal) [inline Samples/Etg.Yams.ARM/Etg.Yams.Host/Program.cs]: Extract out of the loop and process first for index 0 then loop from 1 to length

File: Samples/Etg.Yams.ARM/Etg.Yams.Host/Program.cs [hunk 2]
```diff
@@ -104,4 +114,50 @@ public static class YamsConfigBuilderExtensions
...
+                        {
+                            // program name is the first argument
+                            programName = arguments[i];
+                        }
+                        else
+                        {
+                            // Remove these servicename and instance arguments as TopShelf adds them as well
+                            // Remove install switch
+                            if (arguments[i].StartsWith("-servicename", StringComparison.InvariantCultureIgnoreCase) |
```
Comment (nehmebilal) [inline Samples/Etg.Yams.ARM/Etg.Yams.Host/Program.cs]: This repeated line could be extracted to a private function

File: Samples/Etg.Yams.ARM/Etg.Yams.Host/Program.cs [hunk 3]
```diff
@@ -104,4 +114,50 @@ public static class YamsConfigBuilderExtensions
...
+                            {
+                                continue;
+                            }
+                            argumentsList.Append(" ");
+                            argumentsList.Append(arguments[i]);
+                        }
+                    }
+
+                    // Apply the arguments to the ImagePath value under the service Registry key
+                    var imageName = $"\"{Environment.CurrentDirectory}\\{programName}\" {argumentsList}";
```
Comment (nehmebilal) [inline Samples/Etg.Yams.ARM/Etg.Yams.Host/Program.cs]: What is is for?

**Comment by mhertis** [comment]
When Yams is installed as WindowsService with TopShelf the command line arguments used with installation were not passed to the service at startup. As result we have had installed Windows Service that could not be started because of missing arguments.

It is known limitation/issue.
https://stackoverflow.com/questions/32892760/set-service-start-parameter-using-topshelf

**Comment by nehmebilal** [comment]
Looks good!

**Pull request merged**